### PR TITLE
[GEOS-6801] Added improved naming and description details for KML network links

### PR DESCRIPTION
**Pull request by gisjon**
Added visibility and interval refresh support to KML as new format_options parameters
Added handling for layer labels and descriptions in SuperOverlay.  Added unit test coverage for all new capabilities. Added explanation of new format_options parameters to vendor parameters page

https://jira.codehaus.org/browse/GEOS-6801

### CODE REVIEW

**Comment by NielsCharlier** [comment]
Looks good to me.

**Comment by jodygarnett** [comment]
We have one GeoServer failure with respect to KML: GeoSearchIntegrationTest
```
expected:<[cite:]BasicPolygons> but was:<[]BasicPolygons>
```

The test code is:
```
assertXpathEvaluatesTo("cite:BasicPolygons",
          "/kml:kml/kml:Document/kml:NetworkLink/kml:name", kml);
```

Would this fix of accidentally started ignoring namespace prefix in kml output?

File: src/kml/src/main/java/org/geoserver/kml/builder/SimpleNetworkLinkBuilder.java [hunk 1]
```diff
@@ -68,8 +73,20 @@ void encodeDocumentContents(Document container) {
         for (int i = 0; i < layers.size(); i++) {
             MapLayerInfo layerInfo = layers.get(i);
             NetworkLink nl = container.createAndAddNetworkLink();
-            nl.setName(layerInfo.getName());
-            nl.setVisibility(true);
+            nl.setName(layerInfo.getLabel());
```
Comment (jodygarnett) [inline src/kml/src/main/java/org/geoserver/kml/builder/SimpleNetworkLinkBuilder.java]: This line is probably the change that created the failure (based on cite:BasicPolygons vs BasicPolygons)

**Comment by jodygarnett** [comment]
@smithkm kindly looked up the spec: https://developers.google.com/kml/documentation/kmlreference#name

TLDR: It is intended to be human readable
